### ISSUE TITLE
Disable figures being opened fullscreen by dsPlot

### ISSUE THREAD

**Issue by kupiqu** [DynaSim/DynaSim#226]
Not sure why this happens, but my guess is that this may happen in some systems when the size of a window is larger than the screen size.

In any case, we should avoid this annoying behavior because it is very  intrusive.

If we are using a matlab option to make figures being opened fullscreen we should remove it. If it is because the size, we should limit it by checking the screen size beforehand in matlab.
 
What do you think?

**Comment by DynaSim** [comment]
The figure size is being set at line 445 in dsPlot:

handles(end+1)=figure('units','normalized','outerposition',[0 0
options.figwidth, options.figheight],'visible',options.visible);

where figwidth and figheight default to 1. This should set the outer
position of the figure to 100% of the screen width and height. That is the
behavior I get on the systems I use. On what systems does this not work?

If normalized units work differently on some systems, perhaps we should use
a fixed size measured in absolute units (e.g., cm)...

**Comment by kupiqu** [comment]
It may be that this is working as expected but given that there are dock menues, or task bars, or etc, the available size is less than the screen size and then under these conditions the window manager (kwin from kde in my ase) maximazes the window.

Btw, I actually meant maximized instead of fullscreen...

**Comment by benpolletta** [comment]
Salva, are you running this on a Mac? What do you mean by "maximized"? What's intrusive about it, and what behavior would you prefer? This has never bothered me, but we could have DynaSim create figures that are a set size (as Jason suggests), which is what Matlab itself does.

**Comment by kupiqu** [comment]
I use Linux. Maximized is that they occupy the whole screen (Mac green button). It's intrusive in the sense that they don't only steal the focus but also hide the rest of the windows in the desktop.

So imagine that you are doing things while Matlab is running, e.g. writing an email or editing some text, or whatever. then Matlab opens a figure as maximized, so you need to restore the size of the figure to be able to continue working on what you were doing.

I don't know for others but in general this may break some people's workflow (including mine).

**Comment by benpolletta** [comment]
Ah, I see. Yeah, I generally have that problem with Matlab - when it plots
a figure, it grabs control of the UI (if that's the right term). I agree,
it can be very annoying. Apparently, if you use this before plotting any
figure it will keep it docked:

set(0,'DefaultFigureWindowStyle','docked')

There's also a file on the File Exchange that's supposed to handle this (no
pun intended):

function h = sfigure(h)
% SFIGURE  Create figure window (minus annoying focus-theft).
%
% Usage is identical to figure.
%
% Daniel Eaton, 2005
%
% See also figure

if nargin>=1
	if ishandle(h)
		set(0, 'CurrentFigure', h);
	else
		h = figure(h);
	end
else
	h = figure;
end


So, we could probably figure out a way to implement something like
this, but I don't know if we want to change DynaSim's behavior or let
users do this themselves. My feeling is it's probably better/simpler
design to have DynaSim replicate Matlab's default behavior.

**Comment by kupiqu** [comment]
Well, there are two things, focus theft (which may be matlab's default, although there are systems like Plasma KDE that allow you to control for that) and figures being maximized, which is what this issue is primarily about, and that is not Matlab's default but a current setting in DynaSim, as far as I can say.

Most likely nobody explicitly set the figure to be maximized in the first place, but I believe the code specified instead a size that is higher than the screensize in some systems (like my laptop), and then the window manager (Kwin in my case) 'smartly' decides that it cannot open such a big window but maximizes it instead (actually a window higher than the screen would be even worse, as surely there would be no way to move or resize the window, so it is indeed smart).

In my opinion, there 3 ways to deal with this:

1- we do nothing
2- we just keep Matlab's default size
3- we set a limit for how big a figure can be based on screensize

Given that the size of a figure has a purpose (to visualize better a multipanel figure), I am more in favor of option 3, because it doesn't maximize windows (which I still consider very intrusive), but on the other hand, it doesn't generate a tiny figure for multipanel plots (matlab's default size).

I don't think this would be so difficult to implement, we just need to do sth like:

```
set(0,'units','pixels');
screenSize = get(0,'screensize');
```

and then control that the actual size is not larger than a fraction fo height and width (whichever we decide the fraction should be).

**Comment by benpolletta** [comment]
Yeah, that sounds good. We could make it 0.9 of screen size, or something.
But I don't see how this will be less intrusive than a figure that grabs
focus but is smaller.

**Comment by erik-roberts** [comment]
we could also have the figure not take focus by setting it to hidden as default and then unhiding it in a way that doesnt move focus i think. we could possibly bring the matlab command window to the front too.

**Comment by kupiqu** [comment]
I would just go simple here. This (focus stealing by figures) is sth users may want or not, and they can configure that on their side.

The issue here is that figures get maximized, at least in my laptop, which is undesirable...

**Comment by kupiqu** [comment]
For this I would just set a maximum size for a figure that would be let's say up to 75% of the size of the screen it will be shown in. So respecting the proportion of the figure, that imposes a limitation based on the first boundary is crossed, either the width or height...

**Comment by erik-roberts** [comment]
one issue with the figure size thing is that for saving through plot functions, one often wants the full size. it would be nice to have an option for figure size in the plot functions if there isnt already. also, the labels are often small so it would be nice to ahve control over that too. making issue.

**Comment by kupiqu** [comment]
> one issue with the figure size thing is that for saving through plot functions, one often wants the full size. it would be nice to have an option for figure size in the plot functions if there isnt already. also, the labels are often small so it would be nice to ahve control over that too. making issue.

plot2svg may be a way to deal with this (to some degree at least)

there are features that are yet missing, but I plan to work on them when I have a chance...
